lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -85,13 +85,45 @@ class AwsCompileCloudFrontEvents {
       properties: {
         AllowedMethods: {
           oneOf: [
-            { enum: ['GET', 'HEAD'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS'] },
-            { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 7,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS', 'PUT', 'PATCH', 'POST', 'DELETE'] },
+            },
           ],
         },
         CachedMethods: {
-          oneOf: [{ enum: ['GET', 'HEAD'] }, { enum: ['GET', 'HEAD', 'OPTIONS'] }],
+          oneOf: [
+            {
+              uniqueItems: true,
+              minItems: 2,
+              items: { enum: ['GET', 'HEAD'] },
+            },
+            {
+              uniqueItems: true,
+              minItems: 3,
+              items: { enum: ['GET', 'HEAD', 'OPTIONS'] },
+            },
+          ],
```

Comment:
In original PR I suggested it to be as:

```javascript
AllowedMethods: {
  enum: [['GET', 'HEAD'], ['GET', 'HEAD', 'OPTIONS'], ...]
}
```

https://github.com/serverless/serverless/pull/8250#discussion_r490242639

I have no idea how it went other way, and why I accept it :)

This is also way simpler form as proposed now. I believe I tested above proposed approach locally. Can you confirm it works and update to that?

Reply:
Yes, it works for the standard cases! However then for instance `AllowedMethods: ['HEAD', 'GET']` is not valid since the order matters. But maybe we can live with that?

Reply:
@jede very good point. I didn't realize that. In light of that, I think it's better if we stick to what you've originally proposed